networkit/graphtools.pyx

```diff
@@ -321,21 +323,61 @@ cdef class GraphTools:
 		return density(graph._this)
 
 	@staticmethod
-	def volume(Graph graph):
+	def volume(Graph graph, nodes = None):
 		"""
-		Get the volume of the input graph.
+		Get the volume (for all outgoing edges) of a graph. If a list of nodes of the graph
+		is given, the volume for the corresponding subgraph is computed.
 
 		Parameters:
 		-----------
 		graph : networkit.Graph
 			The input graph.
+		nodes : list
+			(Optional) List of nodes from the graph.
 
 		Returns:
 		--------
 		double
-			The volume of the input graph.
+			The volume of the subgraph.
 		"""
-		return volume(graph._this)
+
+		cdef vector[node] cNodes
+
+		if nodes is not None:
+			try:
+				cNodes = <vector[node]?>nodes
+				return volume[vector[node].iterator](graph._this, cNodes.begin(), cNodes.end())
+			except TypeError:
+				raise RuntimeError("Error, nodes must be a list of nodes.")
+		else:
+			return volume(graph._this)
+
+	@staticmethod
+	def inVolume(Graph graph, nodes = None):
```

Comment:
If for `inVolume`  the nodes parameter is not optional, why is it declared with a default value of `None`, which just raises an exception below?

Reply:
True. Thanks for all the input and being more careful than me.